whisk/client.go

```diff
@@ -301,7 +301,13 @@ func (c *Client) NewRequest(method, urlStr string, body interface{}, includeName
 }
 
 func (c *Client) addAuthHeader(req *http.Request, authRequired bool) error {
-	if c.Config.AuthToken != "" {
+	// Allow for authorization override via Additional Headers
+	additionalHeaderAuth := c.Config.AdditionalHeaders.Get("Authorization")
```

Comment:
maybe `additionalHeaderAuth` -> `authHeaderValue`

then something like
```
if ! authHeaderValue {
  encodedAuthToken := base64.StdEncoding.EncodeToString([]byte(c.Config.AuthToken))
  authHeaderValue = fmt.Sprintf("Basic %s", encodedAuthToken)
}
req.Header.Add("Authorization", authHeaderValue )
```

Reply:
I am attempting to preserve the current checks for `c.Config.AuthToken != ""` is that something you'd like me to remove entirely? I don't think this logic can be condensed much without getting rid of that check

Reply:
also is `!authHeaderValue` equivalent to `authHeaderValue == ""`?

Reply:
we'll need to keep something equivalent... basically
1. only set the auth header if a auth header value is supplied
2. the auth header value can come from either the c.Config.AuthToken or the c.Config.AdditionalHeaders
3. if the auth header value comes from c.Config.AdditionalHeaders, always use that one
4. i think only one line is needed to add the auth header.  no need to remove it.

Reply:
```if ! authHeaderValue``` is a lazy psuedo code for checking that `c.Config.AdditionalHeaders.Get("Authorization")` actually returns a value (i.e. not nil, not empty string)

Reply:
Ok i think I got what you want - just needed to think it through